Assets/MRTK/SDK/Features/UX/Scripts/Pointers/TeleportPointer.cs

```diff
@@ -231,6 +246,59 @@ protected Gradient GetLineGradient(TeleportSurfaceResult targetResult)
             }
         }
 
+        /// check if a backstrafe is possible on a valid platform regarding the possible strafe height given
+        /// </summary>
+        /// <param name="newPosition">the new position relative to backstrafe position</param>
+        /// <param name="hitStrafePosition">actual position the strafe raycast hits</param>
+        /// <returns>if there is a valid layer one can backstrafe on</returns>
+        internal bool CheckPossibleBackStep(Vector3 newPosition, out Vector3 hitStrafePosition)
+        {
+            var raycastProvider = CoreServices.InputSystem.RaycastProvider;
+            Vector3 strafeOrigin = new Vector3(newPosition.x, MixedRealityPlayspace.Position.y + maxHeightChangeOnStrafe, newPosition.z);
+            Vector3 strafeTerminus = strafeOrigin + (Vector3.down * maxHeightChangeOnStrafe * 2f);
+            Debug.Log("origin" + strafeOrigin);
+            Debug.Log("terminus" + strafeTerminus);
+
+            RayStep rayStep = new RayStep(strafeOrigin, strafeTerminus);
+            LayerMask[] layerMasks =  new LayerMask[] { ValidLayers };
+
+            // check are we hiting a floor plane or step above the current MixedRealityPlayspace.Position
+            if (raycastProvider.Raycast(rayStep, layerMasks, false, out var hitInfo))
+            {
+                hitStrafePosition = hitInfo.point;
+                return true;
+            }
+
+            hitStrafePosition = Vector3.zero;
+            return false;
+        }
+
+        /// <summary>
+        /// Performs a strafe in the opposite direction of the camera's forward direction
+        /// </summary>
+        internal void PerformStrafe()
+        {
+            canMove = false;
+            var height = MixedRealityPlayspace.Position.y;
+            var newPosition = -CameraCache.Main.transform.forward * strafeAmount + MixedRealityPlayspace.Position;
+
+            newPosition.y = height;
+            bool isValidStrafe = true;
+            if (checkForFloorOnStrafe)
+            {
+                isValidStrafe = CheckPossibleBackStep(newPosition, out var strafeHitPosition);
+                if (adjustHeightOnStrafe)
```

Comment:
Do we want to adjust the height even if the strafe is not valid?

Reply:
Good catch I'll fix this

Reply:
[Updated the comment after looking into this]